scipy/ndimage/_measurements.py

```diff
@@ -618,77 +622,76 @@ def _stats(input, labels=None, index=None, centered=False):
     """
     def single_group(vals):
         if centered:
-            vals_c = vals - vals.mean()
-            return vals.size, vals.sum(), (vals_c * vals_c.conjugate()).sum()
+            vals_c = vals - xp.mean(vals)
+            return vals.size, xp.sum(vals), xp.sum(vals_c * vals_c.conjugate())
         else:
-            return vals.size, vals.sum()
+            return vals.size, xp.sum(vals)
 
     if labels is None:
         return single_group(input)
 
     # ensure input and labels match sizes
-    input, labels = np.broadcast_arrays(input, labels)
+    input, labels = xp.broadcast_arrays(input, labels)
 
     if index is None:
         return single_group(input[labels > 0])
 
-    if np.isscalar(index):
+    if xp_isscalar(index, xp):
         return single_group(input[labels == index])
 
     def _sum_centered(labels):
         # `labels` is expected to be an ndarray with the same shape as `input`.
         # It must contain the label indices (which are not necessarily the labels
         # themselves).
-        means = sums / counts
-        centered_input = input - means[labels]
+        means = xp.astype(sums, xp.float64) / xp.astype(counts, xp.float64)   # XXX: f64 hardcoded
+        centered_input = xp.astype(input, means.dtype) - xp.take(means, labels)
```

Comment:
This one is a fun one:  in numpy, `means[labels]`, is defined for `labels.ndim > 1` (in a weird way but hey); array-api-strict seems to only allow 1D labels.